sys/net/link_layer/ieee802154/ieee802154.c

```diff
@@ -27,6 +27,14 @@ size_t ieee802154_set_frame_hdr(uint8_t *buf, const uint8_t *src, size_t src_len
     uint8_t type = (flags & IEEE802154_FCF_TYPE_MASK);
     uint8_t bcast = (flags & IEEE802154_BCAST);
 
+    if (dst_pan.u16 == src_pan.u16 &&
+        (src_len != 0 && dst_len != 0)) {
```

Comment:
@miri64, this is no driver code (I thought)?! I was talking about these 2 files

1. /sys/net/link_layer/ieee802154/ieee802154.c
2. /sys/net/gnrc/link_layer/netdev2/gnrc_netdev2_ieee802154.c

both are not driver, but `src` is set in the GNRC (2.) part but `dst` if broad/multicast is set in the other file (2.). I argue for consistency, i.e., set broadcast address in GNRC too, and further get rid of this broadcast flag setting and later unsetting again.